vividus-http-client/src/main/java/org/vividus/http/client/ResponseValidatingHttpClient.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2019-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.http.client;
+
+import java.io.IOException;
+
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.protocol.HttpContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ResponseValidatingHttpClient extends HttpClient
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(ResponseValidatingHttpClient.class);
+
+    @Override
+    public HttpResponse execute(HttpUriRequest request, HttpContext context) throws IOException
+    {
+        HttpResponse response = super.execute(request, context);
+        int status = response.getStatusCode();
+        if (status >= HttpStatus.SC_OK && status < HttpStatus.SC_MULTIPLE_CHOICES)
```

Comment:
How one could overcome the case when he needs something above 300 to be considered as acceptable SC?
In this case, do we need to have 2 HttpClients ? 
Maybe the validation part could be configuratble?

Reply:
agree in general, but not sure whether it's worth implementing now since we haven't had such requests yet (for years), I mean the chance is too low

Reply:
okay :)